src/App.tsx

```diff
@@ -1,8 +1,58 @@
 import React from 'react';
 import './App.css';
+import { getGoods } from './GoodsApi';
+import { GoodsList } from './components/GoodsList';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface MyState {
+  goods: Good[];
+}
+
+interface Good {
+  id: number;
+  name: string;
+  color: string;
+}
+
+class App extends React.Component<{}, MyState> {
+  state: MyState = {
+    goods: [],
+  };
+
+  handleClickAllGoods = () => {
+    getGoods().then(goods => this.setState({
+      goods,
+    }));
+  };
+
+  handleClickFirstFive = () => {
+    getGoods().then(goods => this.setState({
+      goods: goods
+        .sort((a: Good, b: Good) => a.name.localeCompare(b.name))
+        .splice(0, 5),
+    }));
+  };
+
+  handleClickOnlyRed = () => {
+    getGoods().then(goods => this.setState({
+      goods: goods.filter((good: Good) => good.color === 'red'),
```

Comment:
опиши что отдает функция и это не нужно будет
```suggestion
      goods: goods.filter((good) => good.color === 'red'),
```